django/db/models/fields/related.py

```diff
@@ -1586,7 +1587,20 @@ class ForeignObject(RelatedField):
 
     def __init__(self, to, from_fields, to_fields, rel=None, related_name=None,
             related_query_name=None, limit_choices_to=None, parent_link=False,
-            on_delete=CASCADE, swappable=True, **kwargs):
+            on_delete=None, swappable=True, **kwargs):
```

Comment:
It could be moved to right after `from_fields` and `to_fields` (it can't go before them because they are actually required and have no default). But IMO there's not much value in this; `ForeignObject` is not public API, and almost nobody (except maybe Django developers, or someone working on some custom type of relational field) types its signature directly.

But because it's not public API we're also free to rearrange its signature as we see fit, with no need for a deprecation path, so I'd be fine with moving `on_delete` earlier in the signature.